packages/core/src/provider/rpcProvider.ts

```diff
@@ -0,0 +1,48 @@
+import { HttpProvider } from "./httpProvider";
+import { RPCError } from "./error/rpcError";
+import { IRequestOptions } from "./types";
+
+interface IRPCRequest {
+    jsonrpc: string;
+    id: number;
+    method: string;
+    params?: any[];
+}
+
+export class RPCProvider extends HttpProvider {
```

Comment:
I think we will prob wrap a starknet.js rpc provider here right? Rather than extend HttpProvider?